Bottom: 80d2b2ad221180dcd6e6aba0f482246576de4d34
Top:    bcd5cdb833e385404b5d9f7a62d720a03c1231b6
Author: Jason Wang <jasowang@redhat.com>
Date:   2017-07-25 16:45:31 +0800

Refresh of tun-xdp-support

---

diff --git a/drivers/net/tun.c b/drivers/net/tun.c
index 1645f42..3c9e60e 100644
--- a/drivers/net/tun.c
+++ b/drivers/net/tun.c
@@ -1273,7 +1273,9 @@ static void tun_xdp_xmit(struct tun_struct *tun, struct tun_file *tfile,
 	int ret;
 
 	skb->queue_mapping = tfile->queue_index;
+	local_bh_disable();
 	ret = tun_net_xmit(skb, tun->dev);
+	local_bh_enable();
 	if (ret == NET_XMIT_DROP) {
 		this_cpu_inc(tun->pcpu_stats->tx_dropped);
 		kfree_skb(skb);
